test/unit-test/FreeRTOS_ARP/FreeRTOS_ARP_utest.c

```diff
@@ -1553,48 +1550,51 @@ void test_eARPGetCacheEntry_IPMatchesOtherBroadcastAddr( void )
     /* =================================================== */
 }
 
-void test_eARPGetCacheEntry_LocalIPIsZero( void )
-{
-    uint32_t ulIPAddress;
-    MACAddress_t xMACAddress;
-    eARPLookupResult_t eResult;
-    uint32_t ulSavedGatewayAddress;
-    struct xNetworkInterface * xInterface;
-    struct xNetworkEndPoint * pxEndPoint, xEndPoint;
-
-    /* =================================================== */
-    *ipLOCAL_IP_ADDRESS_POINTER = 0;
-    ulIPAddress = 0x1234;
-    /* Not worried about what these functions do. */
-    FreeRTOS_FindEndPointOnIP_IPv4_ExpectAnyArgsAndReturn( NULL );
-    xIsIPv4Multicast_ExpectAndReturn( ulIPAddress, 0UL );
-    FreeRTOS_FindEndPointOnNetMask_ExpectAndReturn( ulIPAddress, 4, ( void * ) 0x1234 );
-    eResult = eARPGetCacheEntry( &ulIPAddress, &xMACAddress, &pxEndPoint );
-    TEST_ASSERT_EQUAL_MESSAGE( eCantSendPacket, eResult, "Test 4" );
-    /* =================================================== */
-}
-
-void test_eARPGetCacheEntry_LocalIPMatchesReceivedIP( void )
-{
-    uint32_t ulIPAddress;
-    MACAddress_t xMACAddress;
-    eARPLookupResult_t eResult;
-    uint32_t ulSavedGatewayAddress;
-    struct xNetworkInterface * xInterface;
-    struct xNetworkEndPoint * pxEndPoint, xEndPoint;
-
-    /* =================================================== */
-    *ipLOCAL_IP_ADDRESS_POINTER = 0x1234;
-    ulIPAddress = *ipLOCAL_IP_ADDRESS_POINTER;
-    /* Not worried about what these functions do. */
-    xEndPoint.ipv4_settings.ulIPAddress = 0x1234;
-    FreeRTOS_FindEndPointOnIP_IPv4_ExpectAnyArgsAndReturn( &xEndPoint );
-    xIsIPv4Multicast_ExpectAndReturn( ulIPAddress, 0UL );
-    FreeRTOS_FindEndPointOnNetMask_ExpectAndReturn( ulIPAddress, 4, ( void * ) 0x1234 );
-    eResult = eARPGetCacheEntry( &ulIPAddress, &xMACAddress, &pxEndPoint );
-    TEST_ASSERT_EQUAL_MESSAGE( eARPCacheHit, eResult, "Test 5" );
-    /* =================================================== */
-}
+// TODO: _TJ_: the following 2 tests are commented out for now as the required changes in the
```

Comment:
A comment can be added saying that initial analysis shows that the tests are not valid for endpoint based implementation. However, it will be re-evaluated and if found valid will be added as a part of a new PR.